### PR TITLE
Implement the FXCop Avoid Empty Interfaces rule

### PR DESCRIPTION
**Pull request by rchande**

### CODE REVIEW

**Comment by rchande** [comment]
@srivatsn Who should I tag for review?

**Comment by srivatsn** [comment]
@mavasani @lgolding @heejaechang @JohnHamby @genlu

File: src/Microsoft.ApiDesignGuidelines.Analyzers/Core/AvoidEmptyInterfaces.cs [hunk 1]
```diff
@@ -33,7 +35,17 @@ public abstract class AvoidEmptyInterfacesAnalyzer : DiagnosticAnalyzer
...
-            
+            analysisContext.RegisterSymbolAction(AnalyzeInterface, SymbolKind.NamedType);
+        }
+
+        private void AnalyzeInterface(SymbolAnalysisContext context)
+        {
+            var symbol = (INamedTypeSymbol)context.Symbol;
+            if (symbol.TypeKind == TypeKind.Interface && !symbol.GetMembers().Any())
+            {
+                var declaration = symbol.DeclaringSyntaxReferences.First();
```
Comment (srivatsn) [inline src/Microsoft.ApiDesignGuidelines.Analyzers/Core/AvoidEmptyInterfaces.cs]: Unneccessary?

File: src/Microsoft.ApiDesignGuidelines.Analyzers/Core/AvoidEmptyInterfaces.cs [hunk 2]
```diff
@@ -33,7 +35,17 @@ public abstract class AvoidEmptyInterfacesAnalyzer : DiagnosticAnalyzer
 
         public override void Initialize(AnalysisContext analysisContext)
         {
-            
+            analysisContext.RegisterSymbolAction(AnalyzeInterface, SymbolKind.NamedType);
+        }
+
+        private void AnalyzeInterface(SymbolAnalysisContext context)
```
Comment (srivatsn) [inline src/Microsoft.ApiDesignGuidelines.Analyzers/Core/AvoidEmptyInterfaces.cs]: Looking at the FxCop implementation, it seems to not fire on interfaces that have method through their base interfaces. I,e `IFoo` itself is empty but `IFoo` derives from `IBar` which has some methods. Can you cover that case as well?

**Comment by mavasani** [comment]
Can you also fix the descriptor to set isEnabledByDefault to true and include the MSDN help URL for this diagnostic?

**Comment by rchande** [comment]
Thanks guys, I've implemented your feedback.

**Comment by genlu** [comment]
:+1:

**Comment by lgolding** [comment]
@genlu Do you want to merge it?

**Comment by rchande** [comment]
@genlu I'm going to go ahead and merge this.

**Pull request merged**